components/admin_console/permission_schemes_settings/permissions_tree/permissions_tree.jsx

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import {FormattedMessage} from 'react-intl';
+
+import PermissionGroup from '../permission_group.jsx';
+
+import EditPostTimeLimitButton from '../edit_post_time_limit_button';
+import EditPostTimeLimitModal from '../edit_post_time_limit_modal';
+
+export default class PermissionsTree extends React.Component {
+    static propTypes = {
+        scope: PropTypes.string.isRequired,
+        config: PropTypes.object.isRequired,
+        role: PropTypes.object.isRequired,
+        onToggle: PropTypes.func.isRequired,
+        parentRole: PropTypes.object,
+        selected: PropTypes.string,
+        selectRow: PropTypes.func.isRequired,
+        readOnly: PropTypes.bool,
+    };
+
+    static defaultProps = {
+        role: {
+            permissions: [],
+        },
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            editTimeLimitModalIsVisible: false,
+        };
+
+        this.ADDITIONAL_VALUES = {
+            edit_post: {
+                editTimeLimitButton: <EditPostTimeLimitButton onClick={() => this.onClickEditPostTimeLimitButton()}/>,
```

Comment:
Why use an arrow function here?

Reply:
Actually, It wasn't part of my change. I can change it to use directly this.onClickEditPostTimeLimitButton. @mkraft any concern about it?

Reply:
Done